src/runtime/cmd/kata-runtime/kata-check_amd64_test.go

```diff
@@ -322,6 +325,30 @@ func TestCheckHostIsVMContainerCapable(t *testing.T) {
 	err = hostIsVMContainerCapable(details)
 	assert.Nil(err)
 
+	// Remove required kernel modules and add them to denylist
```

Comment:
I have two remarks here :

1. the "failure" test already removes the files on disk. This means they won't be loadable anymore and that we don't need to bother with a deny list,
2. making sure that the modules are unloaded from memory is rather a requirement of the "failure" test. For clarity, I'd move this block under the `// remove the modules to force a failure` comment.

Reply:
Removing module files is not enough for passing the "failure" test. Let me give a more details about that:

1. `sysModuleDir` is replaced with a temp dir, e.g. "/tmp/tmp.xxGmSePqj2".
2. Call `createModules()`. The function just creates **regular files** to simulate system modules. https://github.com/kata-containers/kata-containers/blob/3b2fb6a604a624f1ec0c21a85d6e4182344aac4c/src/runtime/cmd/kata-runtime/kata-check_test.go#L68-L71
3. Remove `sysModuleDir` under the "// remove the modules to force a failure" comment.
4. Call `haveKernelModule()`. This function checks existence of a kernel module. Three steps are involved:
    1. If a kernel module exists at `sysModuleDir`, return true. In this case, it won't return as the dir was removed. https://github.com/kata-containers/kata-containers/blob/3b2fb6a604a624f1ec0c21a85d6e4182344aac4c/src/runtime/cmd/kata-runtime/kata-check.go#L139-L142
    2. Check if the current user is root. (Our test is started with `sudo`, it won't return here.)
    3. Try to load the module. Returns false if it fails. Please note that if a module was loaded in the memory or wasn't added to blacklist, loading module will be succeed. https://github.com/kata-containers/kata-containers/blob/3b2fb6a604a624f1ec0c21a85d6e4182344aac4c/src/runtime/cmd/kata-runtime/kata-check.go#L152-L156
    4. Return true.
 
I admit adding the blacklist does more dirty work in the test, but I have no idea how to pass the test without this.

Reply:
Plus, the blacklist file is restored after the function is completed. It won't poison environment.